### DISCUSSION TITLE
invoice page break after specific line item

### DISCUSSION THREAD

**Discussion by McP4nth3r** [QuestPDF/QuestPDF#993] [Q&A]
Hey, i would like to have a page break after a specific line item. I've modified the invoice with subtotal and now i need to break the page after a line item with property "ForcePageBreak" when I build a site with PageBreak then an error appears "Dynamic Content does not fit page" or similar. So everybody have an idea how to realize this behavior? :)

**Comment by MarcinZiabek** [comment]
The DynamicComponent requires you to split the content and generate a separate layout for each page. Once you determine which items should be visible on the page, you don't need to add the `PageBreak`.

Please use the following example as a source of inspiration that shows how to divide many items into multiple pages with per-page totals. Your use case is likely much easier to do: https://www.questpdf.com/concepts/dynamic-components.html#table-with-per-page-totals

**Reply by McP4nth3r** [reply]
Ok this works fine thanks! :)
next problem :D
when I have a line item that contains many text then the text should be splitted instead of jump  to next page with enough space. When the space isn't enough on any page then nothing is displayed.

**Reply by girlpunk** [reply]
Are you able to share some more information on why `DynamicComponent` is required in your use case?

**Reply by McP4nth3r** [reply]
yeah of course,
i need a subtotal amount that are calculated for the line items. This means the text line item is part of the line item table. i have line item types 
```cs
public enum ELineItemKind
{
    POSITION = 0,
    TEXT = 1,
    ALTERNATIVE = 2,
    OPTIONAL = 3,
    SHADOW = 4,
    SUBTOTAL = 5,
}
```
the goal is to have a table structure that displays line items like on an invoice but with the ability to add text infront or after a specific position to show instructions or specific informations about an article or a service.

**Reply by girlpunk** [reply]
If the subtotal is for all items up to this point in the document (as opposite to a total for the page or similar), it's likely you can generate this without using a dynamic component.
As an idea of where to start, consider this example:

```csharp
var subtotal = 0.0;
for (var row in lineItems) {
    switch (row.type) {
        case Type.Item:
            table.Cell().Text(Item.Text);
            table.Cell().Text(Item.Value.ToString());
            subtotal += Item.Value;
            break;
        case Type.Subtotal:
            table.Cell().Text("Subtotal:");
            table.Cell().Text(subtotal.ToString());
            subtotal = 0;
            break;
    }
}
```

**Reply by McP4nth3r** [reply]
yeah i know what you mean but how can i get the subtotal for each page? i need it on top the subtotal of the previous page und at the bottom of the page the subtotal of all line items including prev sides displayed to this time

**Reply by girlpunk** [reply]
Ah, apologies I've misunderstood your use case then. In which case I believe this is currently something you would need to implement manually.

**Reply by McP4nth3r** [reply]
i have a while loop that checks if the text is to long than it cuts word for word till it fits. the problem is the compose method is many times invoked to i can't track the already displayed text to display the rest of the text.
